website/database.py

```diff
@@ -728,16 +732,19 @@ def delete_class(self, Class):
     def resolve_class_link(self, link_id):
         return CLASSES.get({"link": link_id})
 
-    def get_username_invite(self, username):
-        return INVITATIONS.get({"username": username}) or None
+    def get_user_class_invite(self, username, class_id):
+        return INVITATIONS.get({"username#class_id": f"{username}#{class_id}"}) or None
+
+    def add_class_invite(self, key, data):
+        INVITATIONS.update({"username#class_id": key}, data)
 
-    def add_class_invite(self, data):
-        INVITATIONS.put(data)
+    def remove_user_class_invite(self, username, class_id):
+        return INVITATIONS.delete({"username#class_id": f"{username}#{class_id}"})
 
-    def remove_class_invite(self, username):
-        INVITATIONS.delete({"username": username})
+    def get_user_invitations(self, username):
+        return INVITATIONS.get_many({"username": username}) or []
```

Comment:
Doesn't `get_many` already return an empty list by itself?

Reply:
Yeah, it returns a ResultPage class which, if iterated over, we iterate over its records; a list indeed! Will change it, thanks!